services/src/main/java/org/exoplatform/addons/gamification/search/RuleSearchConnector.java

```diff
@@ -60,7 +60,9 @@ public class RuleSearchConnector {
       + "            }\n"
       + "          }\n"
       + "        }\n";
-
+  
+  private static final String          ILLEGAL_SEARCH_CHARACTERS    = "\\!?^()+-=<>{}[]:\"'*~&|";
```

Comment:
we may have to espace those characters instead of deleting them

Reply:
or you can reuse the same algorith used here for more coherence : https://github.com/Meeds-io/social/blob/b8a321d0711522990b33d2fa342f9a8557fed9af/component/core/src/main/java/org/exoplatform/social/core/jpa/search/ActivitySearchConnector.java#L317-L321

Reply:
e do escape them :
https://github.com/Meeds-io/gamification/pull/393/files#diff-50e9310b0f0ce4e8a8e1f71e32d23a7b3daf42f7f8a7fdefe852570fdf59e284R289

Reply:
yep, to reuse the same algorithm as mentioned in previous [comment](https://github.com/Meeds-io/gamification/pull/393/files#r914909128)

Reply:
already escaped not deleted :
https://github.com/Meeds-io/gamification/pull/393/files#diff-50e9310b0f0ce4e8a8e1f71e32d23a7b3daf42f7f8a7fdefe852570fdf59e284R289

Reply:
ok, the characters to escape are only https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-simple-query-string-query.html#simple-query-string-query-notes